### PR TITLE
Dev 17

### PR DESCRIPTION
**Pull request by Hoppop**
When creating items with tags that don't exist, create the tags

### CODE REVIEW

File: server/app/services/item_tag.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+from typing import List
+
+from ..models.tag import Tag
+from ..models.user import User
+from ..models.item_tag import ItemTag
+from ..schemas.tag import TagCreate
+from ..database import get_session
+
+
+def create_item_items_tags(item_id: int, tags: List[TagCreate], user: User):
```
Comment (MichaelBabushkin) [inline server/app/services/item_tag.py]: consider changing the function name to something more "Self-explanatory".

File: server/app/services/item_tag.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    tags = [TagCreate(name=user.username)] if not tags else tags
```
Comment (MichaelBabushkin) [inline server/app/services/item_tag.py]: The line tags = [TagCreate(name=user.username)] if not tags else tags could potentially overwrite the input parameter which is generally not a good practice. A separate variable should be used if modifying the list of tags is necessary.

File: server/app/services/item_tag.py (hunk 1 above)
Comment (Hoppop) [reply server/app/services/item_tag.py]: I think it is "Self-explanatory" but I'm open for suggestions

File: server/app/services/item_tag.py (hunk 2 above)
Comment (Hoppop) [reply server/app/services/item_tag.py]: Thanks! Added a fix

File: server/app/services/item_tag.py (hunk 1 above)
Comment (MichaelBabushkin) [reply server/app/services/item_tag.py]: I think 'add_tags_to_item' might be a better option as it describes the functionality more precisely.

Comment (Hoppop) [reply server/app/services/item_tag.py]: But it's not what it does

Comment (Hoppop) [reply server/app/services/item_tag.py]: Added a fix

**Review by MichaelBabushkin — APPROVED** [review]

**Pull request merged**